### PR TITLE
README: document layer structure

### PR DESCRIPTION
**Pull request by mirzak**
Leaving "stealth mode" on this repository :). Please let me know what you think about this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
 
 ![Mender logo](https://mender.io/user/pages/05.resources/06.digital-assets/logo.png)
 
+## Structure
+
+This section is for anyone interested in contribution support for an board.
```
Comment (drewmoseley) [inline README.md]: This section is for anyone interested in contributing support for a board.

File: README.md [hunk 2]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+
+meta-mender-community is a top level layer which is not actually a layer it
```
Comment (drewmoseley) [inline README.md]: meta-mender-community is a repository containing multiple Yocto layers for integrating Mender onto various boards.

File: README.md [hunk 3]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+self but contains a collection of layers which integrates Mender on various
+boards.
+
+The layers are structured based on upstream BSP layers and not individual
+boards.
+
+There are several types of BSP layers, some are SoC oriented and a selection of
```
Comment (drewmoseley) [inline README.md]: There are multiple types of layers included here.

SoC-oriented:
...
Vendor-oriented:
...

File: README.md [hunk 4]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+- meta-renesas
+
+And then you also have System on Module vendors that typically have their own
+BSP layers which append to the SoC ones, and a selection of those would be:
+
+- meta-variscite
+- meta-phytec
+- meta-toradex-nxp
+
+Naming of integration layers in meta-mender-community follows the upstream
```
Comment (drewmoseley) [inline README.md]: Naming of integration layers follows the upstream naming conventions with SoM vendor layer name having priority.  The convention is:

File: README.md [hunk 5]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+naming closely where SoM vendor layer name take precedence and the convention
+is:
+
+   meta-mender-<upstream suffix>
+
+This should make it clear which layer is targeting what BSP.
+
+### Manifest files
+
+We utilize the [google repo](https://source.android.com/setup/develop/repo) tool
```
Comment (drewmoseley) [inline README.md]: The google repo tool and associated manifest files are used for managing the list of repositories needed for these builds. The common manifest (meta-mender-community/scripts/mender.xml) is included by board-specific manifests and include the required Mender layers:

File: README.md [hunk 6]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+
+This includes the required Mender layers,
+[meta-mender](https://github.com/mendersoftware/meta-mender) and
+[meta-mender-community](https://github.com/mendersoftware/meta-mender-community).
+
+Each integration layer should provide a manifest file, e.g
+
+    meta-mender-community/meta-mender-sunxi/scripts/manifest-sunxi.xml
+
+Above includes `mender.xml`
```
Comment (drewmoseley) [inline README.md]: This line can be removed if you accept my suggestion above

File: README.md [hunk 7]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+
+### Templates and configuration fragments
+
+The typical work flow utilizing this layer is,
```
Comment (drewmoseley) [inline README.md]: To use this layer:

1. Download the source:
     ...
2. Setup the local build environment:
     ...
3. Run the build:
     ...

The `setup-environment` script provided is a wrapper for the Yocto `oe-init-build-env` script.

File: README.md [hunk 8]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+    $ . setup-environment <vendor/soc name>
+
+build
+
+    $ bitbake core-image-base
+
+The magic here is the `setup-environment` script which is a wrapper for
+`oe-init-build-env` which you typically call to setup the Bitbake environment.
+
+What `setup-environment` does is:
```
Comment (drewmoseley) [inline README.md]: I think this section can be dropped; they can always read the script if they want this level of detail.

File: README.md [hunk 9]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+
+- source `oe-init-build-env`
+- append common Mender changes to `local.conf`
+    - content of `meta-mender-community/templates/local.conf.append`
+- append integration layer specific changes to `local.conf`
+    - content of `meta-mender-community/meta-mender-<vendor/soc name>/templates/local.conf.append`
+- copy `bblayers.conf` template
+    - `meta-mender-community/meta-mender-<vendor/soc name>/templates/bblayers.conf.sample`
+
+Meaning that each integration layer must provide:
```
Comment (drewmoseley) [inline README.md]: Each integration layer mus provide:
- local.conf.append: this contains board-specific changes to be appended to the Yocto-generated local.conf file.
- bblayers.conf.sample: a templace that will be copied to bblayers.conf

File: README.md [hunk 10]
```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
...
+
+- an `local.conf.append` file containing any board specific changes that are
+required for Mender integration.
+- `bblayers.conf.sample` which is a template that will be copied to
+`bblayers.conf`
+
+### Integration layer examples
+
+Lets start with taking a closer look at meta-mender-qemu:
```
Comment (drewmoseley) [inline README.md]: Is this (all the way to the end) valuable? I think it can be dropped from the readme.

**Comment by mirzak** [comment]
Thanks for the review @drewmoseley, updated based on you suggestions. Left the last one open.

File: README.md (hunk 10 above)
Comment (drewmoseley) [reply README.md]: I think the whole section "Integration Layer Examples" can be removed.

**Pull request merged**